src/libmeasurement_kit/engine.cpp

```diff
@@ -349,12 +683,364 @@ static void task_run(TaskImpl *pimpl, nlohmann::json &settings) {
     // extract and process `options`
     if (settings.count("options") != 0) {
         auto &options = settings.at("options");
-        // TODO(bassosimone): enumerate all possible options. Make sure we
```

Comment:
Replaced this TODO with some autogenerated code (in the following) that starts doing checks.